cfme/physical/physical_server.py

```diff
@@ -68,58 +69,91 @@ def load_details(self, refresh=False):
 
-    def power_off(self):
+    def power_off(self, **kwargs):
+        self._execute_action_button('Power', 'Power Off', **kwargs)
+
+    def power_off_immediately(self, **kwargs):
+        self._execute_action_button('Power', 'Power Off Immediately', **kwargs)
+
+    def restart(self, **kwargs):
+        self._execute_action_button('Power', 'Restart', **kwargs)
+
+    def restart_immediately(self, **kwargs):
+        self._execute_action_button('Power', 'Restart Immediately', **kwargs)
+
+    def refresh(self, provider, handle_alert=False):
+        last_refresh = provider.last_refresh_date()
+        option = 'Refresh Relationships and Power States'
+        self._execute_button('Configuration', option, handle_alert)
+        wait_for(
+            lambda: last_refresh != provider.last_refresh_date(),
+            message="Wait for the server to be refreshed...",
+            num_sec=300,
+            delay=5
+        )
+
+    def lifecycle_provision(self):
         view = navigate_to(self, "Details")
-        view.toolbar.power.item_select("Power Off", handle_alert=True)
+        view.toolbar.lifecycle.item_select("Provision Physical Server")
 
     @variable(alias='ui')
     def power_state(self):
         view = navigate_to(self, "Details")
-        return view.entities.summary("Power Management").get_text_of("Power State")
+        return view.entities.power_management.get_text_of("Power State")
 
     @variable(alias='ui')
     def cores_capacity(self):
         view = navigate_to(self, "Details")
-        return view.entities.summary("Properties").get_text_of("CPU total cores")
+        return view.entities.properties.get_text_of("CPU total cores")
 
     @variable(alias='ui')
     def memory_capacity(self):
         view = navigate_to(self, "Details")
-        return view.entities.summary("Properties").get_text_of("Total memory (mb)")
-
-    def refresh(self, cancel=False):
-        """Perform 'Refresh Relationships and Power States' for the server.
-
-        Args:
-            cancel (bool): Whether the action should be cancelled, default to False
-        """
-        view = navigate_to(self, "Details")
-        view.toolbar.configuration.item_select("Refresh Relationships and Power States",
-            handle_alert=cancel)
+        return view.entities.properties.get_text_of("Total memory (mb)")
 
-    def wait_for_physical_server_state_change(self, desired_state, timeout=300):
+    def _wait_for_state_change(self, desired_state, target, provider, view, timeout=300, delay=10):
         """Wait for PhysicalServer to come to desired state. This function waits just the needed amount of
            time thanks to wait_for.
 
         Args:
             desired_state (str): 'on' or 'off'
+            target (Method): The Method that most be used to compare with the desired_state.
+            view (Navigate): The view that most be refreshed to verify if the value was changed.
```

Comment:
**REQUIRED**
According to the code below `target`is a string, `view` is instance of ":py:class:\`widgetastic.widget.View\`"

Reply:
I forgot to change the doc string.